From 0b9e94639152f3d4f5c13a6ff46087dc51258996 Mon Sep 17 00:00:00 2001
From: Mike Schuchardt <mikes@lunarg.com>
Date: Wed, 11 May 2022 15:43:28 -0700
Subject: [PATCH] cubepp: Fix VK_KHR_display compile error

Vulkan-Hpp Implicit-cast operators on vk::ResultValue were previously
deprecated and finally removed in the v1.3.213 headers. Fix the
resulting compiler error in cubepp by explicitly using the value member
of vk:ResultValue.

Change-Id: Ie441bd56dfb9dfe26e966e0ec3fb3139712a351a
---
 cube/cube.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cube/cube.cpp b/cube/cube.cpp
index 03fd65b46..e984fabb4 100644
--- a/cube/cube.cpp
+++ b/cube/cube.cpp
@@ -2962,7 +2962,7 @@ vk::Result Demo::create_display_surface() {
         exit(1);
     }
 
-    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(display_mode_prop.displayMode, plane_found);
+    vk::DisplayPlaneCapabilitiesKHR planeCaps = gpu.getDisplayPlaneCapabilitiesKHR(display_mode_prop.displayMode, plane_found).value;
     // Find a supported alpha mode
     vk::DisplayPlaneAlphaFlagBitsKHR alphaMode = vk::DisplayPlaneAlphaFlagBitsKHR::eOpaque;
     std::array<vk::DisplayPlaneAlphaFlagBitsKHR, 4> alphaModes = {
